src/mediarithmics/plugins/custom-action/CustomActionBasePlugin.ts

```diff
@@ -0,0 +1,182 @@
+import * as express from "express";
+import * as _ from "lodash";
+
+import {
+  CustomActionRequest,
+  CustomActionPluginResponse,
+  CustomAction,
+} from "./CustomActionInterface";
+import { PropertiesWrapper } from "./../common/BasePlugin";
+import { PluginProperty } from "./../../api/core/plugin/PluginPropertyInterface";
+import { BasePlugin } from "../common";
+
+export interface CustomActionBaseInstanceContext {
+  properties: PropertiesWrapper;
+}
+
+export abstract class CustomActionBasePlugin extends BasePlugin {
+  constructor(enableThrottling = false) {
+    super(enableThrottling);
+
+    this.initCustomAction();
+    this.setErrorHandler();
+  }
+
+  /**
+   *
+   * @param customActionId
+   */
+  async fetchCustomAction(customActionId: string): Promise<CustomAction> {
+    const customActionResponse = await super.requestGatewayHelper(
+      "GET",
+      `${this.outboundPlatformUrl}/v1/scenarios_custom_actions/${customActionId}`
+    );
+    this.logger.debug(
+      `Fetched Custom Action: ${customActionId} - ${JSON.stringify(
+        customActionResponse.data
+      )}`
+    );
+    return customActionResponse.data;
+  }
+
+  /**
+   *
+   * @param customActionId
+   */
+  async fetchCustomActionProperties(
+    customActionId: string
+  ): Promise<PluginProperty[]> {
+    const customActionPropertiesResponse = await super.requestGatewayHelper(
+      "GET",
+      `${this.outboundPlatformUrl}/v1/scenarios_custom_actions/${customActionId}/properties`
+    );
+    this.logger.debug(
+      `Fetched Custom Action Properties: ${customActionId} - ${JSON.stringify(
```

Comment:
Well, I'll stop there, it's always the same remark ;-)

I let you check if there is any other occurrence